src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -0,0 +1,129 @@
+// (C) 2007-2020 GoodData Corporation
+import * as React from "react";
+import get = require("lodash/get");
+import mapboxgl = require("mapbox-gl");
+import { Execution } from "@gooddata/typings";
+import { createPushpinDataLayer } from "./geoChartDataLayers";
+import { createPushpinDataSource } from "./geoChartDataSource";
+import {
+    DEFAULT_DATA_SOURCE_NAME,
+    DEFAULT_LATITUDE,
+    DEFAULT_LONGITUDE,
+    DEFAULT_MAPBOX_OPTIONS,
+    DEFAULT_ZOOM,
+    MAPBOX_ACCESS_TOKEN,
+} from "../../../constants/geoChart";
+import { IGeoConfig, IGeoDataIndex } from "../../../interfaces/GeoChart";
+import { getGeoDataIndex } from "../../../helpers/geoChart";
+
+import "../../../../styles/scss/geoChart.scss";
+
+mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+
+export interface IGeoChartRendererProps {
+    config: IGeoConfig;
+    execution: Execution.IExecutionResponses;
+}
+
+export default class GeoChartRenderer extends React.PureComponent<IGeoChartRendererProps> {
+    public static defaultProps: Partial<IGeoChartRendererProps> = {
+        config: {},
+    };
+
+    private chart: mapboxgl.Map;
+    private chartRef: HTMLElement;
+
+    public componentDidUpdate(prevProps: IGeoChartRendererProps) {
+        const { execution } = this.props;
+        if (execution !== prevProps.execution) {
```

Comment:
I think the result is incorrect when using === operator to compare 2 objects

Reply:
fixed